modules/base/processors/tfselector.h

```diff
@@ -0,0 +1,106 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2018 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#ifndef IVW_TFSELECTOR_H
+#define IVW_TFSELECTOR_H
+
+#include <modules/base/basemoduledefine.h>
+#include <inviwo/core/common/inviwo.h>
+#include <inviwo/core/processors/processor.h>
+#include <inviwo/core/properties/transferfunctionproperty.h>
+#include <inviwo/core/properties/boolproperty.h>
+#include <inviwo/core/properties/optionproperty.h>
+#include <inviwo/core/properties/eventproperty.h>
+#include <inviwo/core/properties/compositeproperty.h>
+#include <inviwo/core/properties/listproperty.h>
+#include <inviwo/core/properties/propertyownerobserver.h>
+#include <inviwo/core/ports/imageport.h>
+
+namespace inviwo {
+
+/** \docpage{org.inviwo.TFSelector, TFSelector}
+ * ![](org.inviwo.TFSelector.png?classIdentifier=org.inviwo.TFSelector)
+ * allows to select a transfer function from a number of presets which can be added and modified by
+ * the user. The image is passed through without any modifications.
```

Comment:
Having an image as input/output is a hack. Should be mentioned that? Does it work without any ports? 
How about encapsulating this into a composite property so that it can be reused?

Reply:
The port is necessary due to the use of event properties in order to cycle through the various TFs using the keyboard. That is the processor needs to be connected to the canvas.

Reply:
Will consider creating a composite property.

Reply:
I also think the setup with the processor is a bit of a hack, should probably be a composite and not a processor, but then we don’t what it in every processor with a tf... do we need some other way of handling this kind of things?

Reply:
I disagree. A composite would definitely come in handy, but in order to provide the TF selection functionality for all other processors featuring a TF (in particular with keyboard shortcuts)